### ISSUE TITLE
Stats not working

### ISSUE THREAD

**Issue by hoan111** [PhlexPlexico/G5API#103]
![image](https://user-images.githubusercontent.com/32768392/121217298-e9a80e80-c8ab-11eb-8e15-af811b18c115.png)

Hello! 
All functions of my G5V and G5API are working properly, except match stats. After I created and started the match, the stats didn't work. G5V only shows "No stats are found for the current match." when the match was stated. I also checked G5API to see if any update requests from my game server have been sent to G5API but none was sent.  This is my debug log from my game server and get5_debuginfo.
Game server: 
![image](https://user-images.githubusercontent.com/32768392/121216367-06901200-c8ab-11eb-8030-e6ad06dad218.png)

get5_debuginfo:
https://pastebin.com/JpRrHRhC
Thank you!

**Comment by PhlexPlexico** [comment]
Hey! Just moving your issue over to G5API since it seems to be an issue with the API moreso than the front-end. 

Are you able to get your logs from the API? If you're running production you should be able to go into the directory and type `yarn pm2 logs G5API --lines 150`. From what I can see, your API is working (navigating shows that it's working). The logs from the API are also located in `~/.pm2/logs` if you want to browse the `G5API-error.log` and the other log file, you may be able to find out what exactly is going wrong.


Thanks!

**Comment by PhlexPlexico** [comment]
Also, what version of the API Plugin are you using on your game server? Just run `sm plugins list` and provide the version for the api_stats plugin.

Thanks!

**Comment by hoan111** [comment]
First, I'm sorry for posting the issue in the wrong position!
The main problem here is when I created a match using your G5V: https://phlex.avidgamers.me/ the playerstats was worked properly. I don't know why.
This is my G5API error log:
![image](https://user-images.githubusercontent.com/32768392/121220253-a602d400-c8ae-11eb-8b12-b80be7580638.png)

API plugin version: 2.0.2 PhlexPlexico version

**Comment by kdev** [comment]
You are trying to connect to a rcon server that does not exist

**Comment by hoan111** [comment]
> You are trying to connect to a rcon server that does not exist

Because of match stats not working so, G5V doesn't know when the match was finished. I turned off the game server before so it appeared that error.

**Comment by PhlexPlexico** [comment]
> 
> 
> First, I'm sorry for posting the issue in the wrong position!
> The main problem here is when I created a match using your G5V: https://phlex.avidgamers.me/ the playerstats was worked properly. I don't know why.
> This is my G5API error log:
> ![image](https://user-images.githubusercontent.com/32768392/121220253-a602d400-c8ae-11eb-8b12-b80be7580638.png)
> 
> API plugin version: 2.0.2 PhlexPlexico version

Did the same server work when you used the public API? Do you have RIPEXT installed on that server? I believe the extension is required in order to make the POST calls. If the game server isn't sending out the POST call (i.e. `/api/match/10/map/0/player/.../update` isn't being called on the API side, there may be some Windows firewall rules that you may have to adjust? I'm not sure, but I would take a stab at it being something network related that isn't allowing the call to go through. 

I suppose the CallBack is only checking for internal server errors (5xx) as well:
https://github.com/PhlexPlexico/get5-webapi/blob/e403387f7daebd7b55066d2b4ef9f8fb6c58b01e/scripting/get5_apistats.sp#L165-L173

So maybe it's unable to even contact the server. Do the vetoes go through as well? If it can only receive values (such as start match through RCON), and not send I feel that there may be a network issue there.

**Comment by hoan111** [comment]
Yes, when I used the same server to contact with public API. And I'm sure that RIPEXT has been installed on my server. For the firewall part, I turned off Windows firewall on my game server and disabled iptables on the API server. Does the problem here come from my API config?

**Comment by hoan111** [comment]
production.json
![image](https://user-images.githubusercontent.com/32768392/121231696-e7997c00-c8ba-11eb-81ad-d74cf96f5d9b.png)

Nginx config:
![image](https://user-images.githubusercontent.com/32768392/121231807-0ef04900-c8bb-11eb-89b1-4619bacc1510.png)

**Comment by PhlexPlexico** [comment]
For your nginx.conf maybe set it to your local IP, technically you don't need to have the API be public (i.e. you shouldn't be able to access `http://g5api.ddns.net:3301` externally). So change that to your local IP instead of the DNS, that's the only difference I can see. 

As for your json file... everything seems to be okay. I would maybe create a match and then just try using Postman and get the API key from the match and try updating stats by yourself as well. See if that works.

**Comment by PhlexPlexico** [comment]
You could also try increasing the debug info for `get5_debug` and check the logs to see if it's being sent. If not, would you mind trying the latest (pending release) build of G5WS?
[G5WS.zip](https://github.com/PhlexPlexico/G5API/files/6619237/G5WS.zip)

**Comment by hoan111** [comment]
I tried your new `get5_apistats` version but it still not working. I both checked Nginx `access.log` and G5API log while veto processing and I realize that no requests from the game server have been sent to Nginx while the debug from the game server said "`Trying to create a request....`" with no errors. I tried to make a manual veto update request to the API server ( using `Postman`) and it works. Also, I updated `REST in PAWN` extension but it still not working.  Here is the log that I got from the game server while veto process, match start is the same,
![image](https://user-images.githubusercontent.com/32768392/121323092-bc09a680-c939-11eb-9d8c-e938bf74639d.png)

**Comment by hoan111** [comment]
I don't know what causes the BUG because I used the same server to create a match from your public API, it worked. My own it didn't work.

**Comment by PhlexPlexico** [comment]
Okay well let's take NGINX and Cloudflare out of the question. Unlike the old webpanel, G5API and the more up to date (latest build on get5 build actions (here)[https://github.com/splewis/get5/suites/2926842702/artifacts/65695247] can use just a public IP instead. This will remove any issues that NGINX or CloudFlare DNS could have.

1) We're going to run a development build of G5V, so first you need to stop your NGINX instance since you're running on port 8080.
2) In the G5V project, navigate to the `vue.config.js` file in the root of the project and change this line to your public IP
https://github.com/PhlexPlexico/G5V/blob/cae3d7d46c423b3e833ae44b7582705b56bf1cb8/vue.config.js#L4
In your case it would be `public: "http://10 3.98.15 2.22 8:8080",`
3) Spin up the development build after saving that file. In that directory just run `yarn serve` and if you navigate to your IP address on port 8080, it should be living. To ensure the API is working as it should, just go to `/api/matches` and see if it loads (preferably in a private window/incognito, due to caching).
4) In your production.json file in G5API, you will also have to change your host and whatnot to your public IP that was listed above. and restart the API.

I've personally never used CloudFlare since I use Namecheap for domain/dns configs. So let's try and take out all of that and move back to basics and see if it works via direct IP connection. 

Also, I noticed in the screenshot you gave me wasn't the plugin that I gave you above, since the debug is still outputting "get5_api" as its log name, and not "G5WS".

Thanks!

**Comment by hoan111** [comment]
ok, this is the G5API log when I spin up the development server. In this case, G5API received requests from the game server to update stats but something goes wrong here.
![image](https://user-images.githubusercontent.com/32768392/121417188-cfe1f680-c993-11eb-95e1-2f113ac0bf19.png)
![image](https://user-images.githubusercontent.com/32768392/121417278-e8eaa780-c993-11eb-944a-02129ac50ee7.png)

> the more up to date (latest build on get5 build actions (here)[https://github.com/splewis/get5/suites/2926842702/artifacts/65695247]
I also installed this get5 version ( this error still appear when I used my previous build )

>  Also, I noticed in the screenshot you gave me wasn't the plugin that I gave you above, since the debug is still outputting "get5_api" as its log name, and not "G5WS".
This "G5WS" plugin that you've sent to me is get5_apistats. Is it true?
![image](https://user-images.githubusercontent.com/32768392/121418639-5ba85280-c995-11eb-8b41-2728483d645f.png)

**Comment by PhlexPlexico** [comment]
The plugin I'm referencing is [here](https://github.com/PhlexPlexico/G5API/files/6619237/G5WS.zip)

That one should be a different plugin at version 2.1.

So when you mean spin up the development server, you mean the front end specifically right? You should still be running the API per normal pm2 starting.

But it seems like the problem may rest in your connectionLimit. Up that to about 50 and give it a try and see if that works? You'll have to restart the API in order for those changes to take effect. But by the looks of it, it appears to being caught up on something during the calls. I would assume it would be a database connection, but that would also fail out and allow you to continue and provide a response time, not just 0ms.

**Comment by hoan111** [comment]
ok. I used Wireshark to capture the packet that has been sent from the game server to the API server and it showed the error "`400 bad request`".  So the problem here comes from the request that has been generated by the` get5_apistats` plugin, specifically by RIPEXT. I updated RIPEXT to the newest version (1.3.0). I'm trying to edit the request made by the `get5_api ` plugin to a valid request. 
![image](https://user-images.githubusercontent.com/32768392/121683729-fadd5f00-cae7-11eb-8073-71f1c032614b.png)

**Comment by PhlexPlexico** [comment]
If you try making the same request to the API it works in Postman I assume? I think I know what one of the issues is. In postman, use this exact URL with the trailing slash.
![image](https://user-images.githubusercontent.com/3514085/121691271-0c8d1b00-ca84-11eb-8f5b-11ebf60b7f86.png)

And let me know if that call fails. I think this may be another issue that I had when trying to do URL downloads, and there may be a way to have NGINX strip that slash as well even with some rule magic.

**Comment by hoan111** [comment]
Every request with the trailing slash is working in Postman. This is my request in Postman
![image](https://user-images.githubusercontent.com/32768392/121722016-190a8580-cb0f-11eb-9583-3aaec4f0f32f.png)
Do `get5_apistats` need to change the way how it makes a request to the API server?

**Comment by PhlexPlexico** [comment]
I was checking it and if you wouldn't mind using 1.2.3 we could see if there's an issue there. That being said, I've always used Linux servers as well, so I could also see this being a quirk of windows servers and how it interacts with cURL requests as well.

**Comment by PhlexPlexico** [comment]
Also, you shouldn't be using x-www-form-encoded when sending via postman, you should select "raw" and make sure it's JSON. That would directly emulate what's being done from G5WS.

**Comment by hoan111** [comment]
I tried with RAW and it's still working. Also, I tried with sm-ripext version 1.2.3. It showed the same result

**Comment by PhlexPlexico** [comment]
Here, let me compile something with a bit more output on the game server response. Regardless of success or failure it should return something to the server console. One moment!

**Comment by PhlexPlexico** [comment]
[G5WS.zip](https://github.com/PhlexPlexico/G5API/files/6640101/G5WS.zip)

There, this should provide some more insight as to why it's failing.

**Comment by hoan111** [comment]
Thank you! I'll try and let you know

**Comment by hoan111** [comment]
This is what I got. Still same before. And your G5WS plugin has an error.
![image](https://user-images.githubusercontent.com/32768392/121732910-fed7a400-cb1c-11eb-94da-a04eae488581.png)
![image](https://user-images.githubusercontent.com/32768392/121732960-1020b080-cb1d-11eb-9786-dbb5af424721.png)
![image](https://user-images.githubusercontent.com/32768392/121732996-1c0c7280-cb1d-11eb-97bb-1900bb47f9e4.png)

**Comment by hoan111** [comment]
Edit: This is a valid request that has been made by Postman. 
![image](https://user-images.githubusercontent.com/32768392/121734492-0bf59280-cb1f-11eb-98f0-6817b82bdf4c.png)

**Comment by PhlexPlexico** [comment]
Seems to be erroring out when it's trying to get a value back on the callback. So I don't think it's hitting the server, since the server only returns JSON objects. There must be something up with the game server at this point. Are you able to test on a Linux server?

**Comment by PhlexPlexico** [comment]
And dumb question, but have you tried compiling G5WS on Windows as well to try and see if it works?

**Comment by PhlexPlexico** [comment]
[G5WS.zip](https://github.com/PhlexPlexico/G5API/files/6640472/G5WS.zip)


Here's a G5WS that is compiled against windows, make sure you're using ripext if you have a windows gameserver as well and meet the requirements for it, too.

**Comment by hoan111** [comment]
> Seems to be erroring out when it's trying to get a value back on the callback. So I don't think it's hitting the server, since the server only returns JSON objects. There must be something up with the game server at this point. Are you able to test on a Linux server for the game server?

Yes. I can test on a Linux server. But I usually use Windows as game server because my public server is running Windows and I use it to host a tournament. I'll test on a linux server later

**Comment by hoan111** [comment]
> And dumb question, but have you tried compiling G5WS on Windows as well to try and see if it works?

You didn't send the G5WS.sp file so I can't complie it :D

**Comment by PhlexPlexico** [comment]
https://github.com/PhlexPlexico/get5-webapi/tree/utility-knife-dmg It's right here :)

**Comment by hoan111** [comment]
![image](https://user-images.githubusercontent.com/32768392/121738549-8d035880-cb24-11eb-974c-02ce0a75428f.png)
Now I got response "Not a correct API key"

**Comment by PhlexPlexico** [comment]
And in your wireshark capture it's the right key?

**Comment by hoan111** [comment]
> And in your wireshark capture it's the right key?

Yes. It's the right key

**Comment by PhlexPlexico** [comment]
[G5WS.zip](https://github.com/PhlexPlexico/G5API/files/6640671/G5WS.zip)

Added in an API Key to print out on the server log when you veto. See what that outputs.

**Comment by hoan111** [comment]
I already made that and complie :D and this is output I got
![image](https://user-images.githubusercontent.com/32768392/121741656-0604af00-cb29-11eb-827d-ffee69d418b4.png)

**Comment by PhlexPlexico** [comment]
Okay so on the API. Right before this line:
https://github.com/PhlexPlexico/G5API/blob/aaba7380256a92f390ddac183114e993ea55c32e/routes/legacy/api.js#L520

Just add in the following `console.log("Our DB Key is: " + matchValues[0].api_key + " and gameserver sent: " + req.body.key);`

I think this may be a case where `localeCompare` is being thrown off because of OS differences. But we shall see :)

**Comment by hoan111** [comment]
This is output I got
![image](https://user-images.githubusercontent.com/32768392/121743153-14ec6100-cb2b-11eb-9238-8e834b448d1a.png)

**Comment by hoan111** [comment]
I hope you can release a stable version for both Windows & Linux CSGO server. Thank for your hard work!

**Comment by PhlexPlexico** [comment]
Last thing to do, can you just output `req.body` as well and let me know what it outputs? I think it's just the windows game server not sending anything proper.

**Comment by hoan111** [comment]
> Last thing to do, can you just output `req.body` as well and let me know what it outputs? I think it's just the windows game server not sending anything proper.
> 
> E: Are you also using NGINX/API on windows?

Sorrry. I've just turned off my computer because too late and I feel sleepy. I'll test and let you know tomorrow :)
At this time, I only use linux (Debian 10) for the API server and Windows (Sometime Linux) for the game server. I think I will not use Nginx/API on Windows in the future

**Comment by PhlexPlexico** [comment]
Oh hey, would you look at that. It seems to be an issue with RipExt. So I would think there may be an issue with RipExt right now. 

https://forums.alliedmods.net/showpost.php?p=2749171&postcount=103

The related fix is here I think? https://github.com/ErikMinekus/sm-ripext/pull/51

It would make sense as to why you're not receiving anything in your body at least. If this is the reason, then you should be receiving data just fine after this is fixed... 

As for right now, maybe just use the latest release of get5_apistats (or build from the master branch and make your debug statements) and use sm-ripext 1.2.3.

**Comment by hoan111** [comment]
> Oh hey, would you look at that. It seems to be an issue with RipExt. So I would think there may be an issue with RipExt right now.
> 
> https://forums.alliedmods.net/showpost.php?p=2749171&postcount=103
> 
> The related fix is here I think? [ErikMinekus/sm-ripext#51](https://github.com/ErikMinekus/sm-ripext/pull/51)
> 
> It would make sense as to why you're not receiving anything in your body at least. If this is the reason, then you should be receiving data just fine after this is fixed...
> 
> As for right now, maybe just use the latest release of get5_apistats (or build from the master branch and make your debug statements) and use sm-ripext 1.2.3.

Yes. I've read that comment in Ailledmodders but not search the fix yet :D So I tell you to check the new version of RIPEXT if it affects to get5_apistats plugin.  Thank you for finding the fix for me :D 

> As for right now, maybe just use the latest release of get5_apistats (or build from the master branch and make your debug statements) and use sm-ripext 1.2.3.

I'll try it and let you know. 
Thank!

**Comment by hoan111** [comment]
UPDATE: I used the latest release of get5_apistats and use sm-ripext 1.2.3 but it's not working. I think I need to install the fix of sm-ripext 1.3.0 but I haven't known how to compile it yet...

**Comment by hoan111** [comment]
UPDATE: I installed the fix of the sm-ripext 1.3.0, then both get5_apistats and G5WS not working ( sending an invalid request to G5API ).

**Comment by hoan111** [comment]
UPDATE:  Everything worked when I using Linux as a game server ( using the latest version of get5_apistats, sm-ripext version 1.2.3).

**Comment by PhlexPlexico** [comment]
So even with the same old plugins, is the API still picking up a blank body?

**Comment by hoan111** [comment]
yes. only Windows appears that error. This error can be caught from API or ripext but I think is API because I tried with the latest build of get5_apistats and both the old version and the latest version of ripext on Windows. It didn't work.

**Comment by PhlexPlexico** [comment]
Okay but I would need more context for this. For example when sending data from the windows game server -> API, can you get the API to `console.log(req.body)` and see if it's undefined? If so, that's undefined behaviour for windows on ripexts part and should be reported there then I believe.

**Comment by hoan111** [comment]
![image](https://user-images.githubusercontent.com/32768392/121819848-9114af00-ccb9-11eb-8c37-cd00242e9487.png)
I added `console.log(req.body)` and restarted the API server but it didn't print the console log.

**Comment by hoan111** [comment]
> ![image](https://user-images.githubusercontent.com/32768392/121819848-9114af00-ccb9-11eb-8c37-cd00242e9487.png)
> I added `console.log(req.body)` and restarted the API server but it didn't print the console log while veto process

**Comment by hoan111** [comment]
![image](https://user-images.githubusercontent.com/32768392/121820541-9d9b0680-ccbd-11eb-851e-1ea775fcb40e.png)
UPDATE: If I use G5WS and ripext version 1.3.0 fixed version, it showed the error above. If I use G5WS and ripext 1.3.0 original release, the `console.log(req.body)` worked. As I can see, the game server didn't send the body to the API server.

**Comment by hoan111** [comment]
![image](https://user-images.githubusercontent.com/32768392/121820780-d1c2f700-ccbe-11eb-9c9a-130fb53dd702.png)
Packet capture

**Comment by ErikMinekus** [comment]
There are debug builds here that print to the server console what ripext is sending and receiving: https://forums.alliedmods.net/showthread.php?p=2749617#post2749617

If I am reading the capture correctly it seems to be sending the request body with a random string (`29536f9800.`) at the beginning and end?

**Comment by hoan111** [comment]
> There are debug builds here that print to the server console what ripext is sending and receiving: https://forums.alliedmods.net/showthread.php?p=2749617#post2749617
> 
> If I am reading the capture correctly it seems to be sending the request body with a random string (`29536f9800.`) at the beginning and end?

Maybe. Because an valid request don't have that string ( Request made by Postman )

**Comment by hoan111** [comment]
> There are debug builds here that print to the server console what ripext is sending and receiving: https://forums.alliedmods.net/showthread.php?p=2749617#post2749617
> 
> If I am reading the capture correctly it seems to be sending the request body with a random string (`29536f9800.`) at the beginning and end?

```
L 06/14/2021 - 11:22:27: get5_event: {
    "matchid": "59",
    "params": {
        "team": "team1",
        "map_name": "de_inferno"
    },
    "event": "map_veto"
}
=== Info: STATE: INIT => CONNECT handle 0x2a3846a0; line 1646 (connection #-5000)

=== Info: Added connection 0. The cache now contains 3051794647169892353 members

=== Info: family0 == v4, family1 == v6

=== Info:   Trying 103.98.152.228:8080...

=== Info: STATE: CONNECT => WAITCONNECT handle 0x2a3846a0; line 1707 (connection #0)

=== Info: Connected to 103.98.152.228 (103.98.152.228) port 8080 (#0)

=== Info: STATE: WAITCONNECT => SENDPROTOCONNECT handle 0x2a3846a0; line 1842 (connection #0)

=== Info: STATE: SENDPROTOCONNECT => DO handle 0x2a3846a0; line 1861 (connection #0)

==> Send header
POST /api/match/59/vetoUpdate HTTP/1.1
Host: 103.98.152.228:8080
User-Agent: sm-ripext/1.3.0
Accept-Encoding: deflate, gzip
Transfer-Encoding: chunked
Accept: application/json
Content-Type: applicaton/json
Expect: 100-continue


=== Info: STATE: DO => DO_DONE handle 0x2a3846a0; line 1935 (connection #0)

=== Info: STATE: DO_DONE => PERFORM handle 0x2a3846a0; line 2056 (connection #0)

=== Info: Mark bundle as not supporting multiuse

=== Info: HTTP 1.1 or later with persistent connection

<== Recv header
HTTP/1.1 100 Continue

==> Send data
298a8e3000
=== Info: Signaling end of chunked upload via terminating chunk.

==> Send data
298a8e3000
=== Info: Mark bundle as not supporting multiuse

=== Info: HTTP 1.1 or later with persistent connection

<== Recv header
HTTP/1.1 500 Internal Server Error

<== Recv header
X-Powered-By: Express

<== Recv header
x-dns-prefetch-control: off

<== Recv header
x-frame-options: SAMEORIGIN

<== Recv header
strict-transport-security: max-age=15552000; includeSubDomains

<== Recv header
x-download-options: noopen

<== Recv header
x-content-type-options: nosniff

<== Recv header
x-xss-protection: 1; mode=block

<== Recv header
access-control-allow-origin: http://103.98.152.228:8080

<== Recv header
vary: Origin

<== Recv header
access-control-allow-credentials: true

<== Recv header
x-ratelimit-limit: 60

<== Recv header
x-ratelimit-remaining: 59

<== Recv header
date: Mon, 14 Jun 2021 04:22:30 GMT

<== Recv header
x-ratelimit-reset: 1623646454

<== Recv header
content-type: application/json; charset=utf-8

<== Recv header
content-length: 36

<== Recv header
etag: W/"24-J/orU2rnoJivUV0ut4rL6VKFSbw"

<== Recv header
set-cookie: G5API=s%3As59b0lT_k3ZvfdBfo3wtd1bGSFFmdxgf.N2oPzXgAYGnI5dkRtd1wzUpUp1nv%2FdbHDFXd5Yh7z%2Fo; Path=/; Expires=Mon, 14 Jun 2021 05:22:30 GMT; HttpOnly

<== Recv header
connection: close

<== Recv header

{"message":"Not a correct API Key."}
<== Recv data
{"message":"Not a correct API Key."}
=== Info: STATE: PERFORM => DONE handle 0x2a3846a0; line 2254 (connection #0)

=== Info: multi_done

=== Info: The cache now contains 3042273941625241600 members

=== Info: Closing connection 0

=== Info: Expire cleared (transfer 0x2a3846a0)

L 06/14/2021 - 11:22:28: [G5WS.smx] [ERR] API request failed, HTTP status code: 500
L 06/14/2021 - 11:22:28: [G5WS.smx] [ERR] Response:
{
    "message": "Not a correct API Key."
}
```

This is what I got when my Windows game server sent a packet to API server. I'll update log in Linux game server soon

**Comment by hoan111** [comment]
```
[get5] Calling Get5_OnEvent(event name = map_veto)
[get5] Calling Get5_OnMapVetoed(team=0, map=de_inferno)
[get5_api] Map Veto START team  map vetoed de_inferno
[get5_api] Our URL is: http://103.98.152.228:8080/api/match/%d/vetoUpdate
[get5_api] Trying to create request to url http://103.98.152.228:8080/api/match/60/vetoUpdate
[get5_api] Accepted Map Veto.
=== Info: STATE: INIT => CONNECT handle 0xea104e0; line 1646 (connection #-5000)

=== Info: Added connection 0. The cache now contains 1 members

=== Info: family0 == v4, family1 == v6

=== Info:   Trying 103.98.152.228:8080...

=== Info: STATE: CONNECT => WAITCONNECT handle 0xea104e0; line 1707 (connection #0)

=== Info: Connected to 103.98.152.228 (103.98.152.228) port 8080 (#0)
=== Info: STATE: WAITCONNECT => SENDPROTOCONNECT handle 0xea104e0; line 1842 (connection #0)

=== Info: STATE: SENDPROTOCONNECT => DO handle 0xea104e0; line 1861 (connection #0)

==> Send header
POST /api/match/60/vetoUpdate/ HTTP/1.1
Host: 103.98.152.228:8080
User-Agent: sm-ripext/1.3.0
Accept-Encoding: deflate, gzip
Transfer-Encoding: chunked
Accept: application/json
Content-Type: application/json
Expect: 100-continue


=== Info: STATE: DO => DO_DONE handle 0xea104e0; line 1935 (connection #0)

=== Info: STATE: DO_DONE => PERFORM handle 0xea104e0; line 2056 (connection #0)

=== Info: Mark bundle as not supporting multiuse

=== Info: HTTP 1.1 or later with persistent connection

<== Recv header
HTTP/1.1 100 Continue

==> Send data
66
{"key": "EOGZN0H8SF1UQQ1WI4V3P1YZ", "map": "de_inferno", "teamString": "team1", "pick_or_veto": "ban"}
9
=== Info: Signaling end of chunked upload via terminating chunk.

==> Send data
0

key": "EOGZN0H8SF1UQQ1WI4V3P1YZ", "map": "de_inferno", "teamString": "team1", "pick_or_veto": "ban"}
9
=== Info: Mark bundle as not supporting multiuse

=== Info: HTTP 1.1 or later with persistent connection

<== Recv header
HTTP/1.1 200 OK

<== Recv header
X-Powered-By: Express

<== Recv header
x-dns-prefetch-control: off

<== Recv header
x-frame-options: SAMEORIGIN

<== Recv header
strict-transport-security: max-age=15552000; includeSubDomains

<== Recv header
x-download-options: noopen
<== Recv header
x-content-type-options: nosniff

<== Recv header
x-xss-protection: 1; mode=block

<== Recv header
access-control-allow-origin: http://103.98.152.228:8080

<== Recv header
vary: Origin

<== Recv header
access-control-allow-credentials: true

<== Recv header
x-ratelimit-limit: 60

<== Recv header
x-ratelimit-remaining: 59

<== Recv header
date: Mon, 14 Jun 2021 05:44:09 GMT

<== Recv header
x-ratelimit-reset: 1623650054

<== Recv header
content-type: application/json; charset=utf-8

<== Recv header
content-length: 21

<== Recv header
etag: W/"15-uFFjCr0SbbbFb/CsC0M2sF++swo"

<== Recv header
set-cookie: G5API=s%3AitUsbeB1MEiQSTyOZexBqessdOxTa4ql.1kLBX%2FV%2BQfCoK8YZs20WZu87kHH%2Fn2QrSYcjDoYGtGE; Path=/; Expires=Mon, 14 Jun 2021 06:44:09 GMT; HttpOnly

<== Recv header
connection: close

<== Recv header

{"message":"Success"
<== Recv data
{"message":"Success"
<== Recv data
}
=== Info: STATE: PERFORM => DONE handle 0xea104e0; line 2254 (connection #0)

=== Info: multi_done

=== Info: The cache now contains 0 members

=== Info: Closing connection 0

=== Info: Expire cleared (transfer 0xea104e0)

```

sm-ripext log in Linux game server

**Comment by PhlexPlexico** [comment]
Also just a heads up, since we're starting to discover this is more of a plugin bug, I'm just going to move the issue to the plugin project :)

Thanks!
